### PR TITLE
RWA-2712 delete API modification

### PR DESCRIPTION
**Pull request by priyankaVerma21**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-2712

### Change description ###
This change is related to check condition in delete API instead of ExclusiveAccess check for privilege access

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by raghera — COMMENTED** [review]
Looks good Priyanka.  Minor comment if fixed can go to QA.

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ management:
         include: health, info, prometheus
 
 config:
-  privilegedAccessClients: ${TASK_MANAGEMENT_PRIVILEGED_CLIENTS:wa_task_management_api,xui_webapp}
+  privilegedAccessClients: ${TASK_MANAGEMENT_PRIVILEGED_CLIENTS:wa_task_management_api,xui_webapp,ccd-case-disposer,ccd_case_disposer}
```
[the remark this answers is not in the document]
Comment (priyankaVerma21) [reply src/main/resources/application.yaml:15]: Just to make sure we have client-id and service name both .

[the remark this answers is not in the document]
Comment (priyankaVerma21) [reply src/main/resources/application.yaml]: Hi @raghera , removed service name from privilegeAccess left ccd-case-disposer and also removed from exclusiveAccess as it's not needed there anymore

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/restrict/ClientAccessControlService.java [hunk 2]
```diff
@@ -46,6 +46,20 @@ public boolean hasPrivilegedAccess(String serviceAuthToken, AccessControlRespons
         return privilegedAccessClients.contains(serviceName);
     }
 
+    /**
+     * Extracts client id from service authorization token and returns if client is whitelisted as exclusiveClient.
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/restrict/ClientAccessControlService.java:50]: Comment should not mention exclusive client as it's confusing.

**Review by raghera — COMMENTED** [review]
There's one comment which should be quick and happy to approve when done to move to QA.

File: src/main/resources/application.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (raghera) [reply src/main/resources/application.yaml]: Thanks.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/restrict/ClientAccessControlService.java (hunk 2 above)
Comment (raghera) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/restrict/ClientAccessControlService.java:50]: @priyankaVerma21 - Have you addressed this one?

**Review by raghera — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/restrict/ClientAccessControlService.java (hunk 2 above)
Comment (priyankaVerma21) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/restrict/ClientAccessControlService.java]: It has been resolved now . Thank you

**Pull request merged**